http3/server.go

```diff
@@ -274,6 +274,15 @@ func (s *Server) ServeListener(ln QUICEarlyListener) error {
 var errServerWithoutTLSConfig = errors.New("use of http3.Server without TLSConfig")
 
 func (s *Server) serveConn(tlsConf *tls.Config, conn net.PacketConn) error {
+	err := s.serveC(tlsConf, conn)
+	// replace quic package specific error with net/http equivalent ones
+	if err == quic.ErrServerClosed {
+		err = http.ErrServerClosed
+	}
+	return err
+}
+
+func (s *Server) serveC(tlsConf *tls.Config, conn net.PacketConn) error {
```

Comment:
Do we need this new function? `quic.ErrServerClosed` can only be returned from `serveListener`, so it should be fine to just apply this logic for the last error returned, no?

Reply:
Yes that was my first change initially.
The new function is more about "future proofing" allowing to catch all return paths and allowing to have one single place to eventually transform more quic errors into http errors if needed later.
up to you to choose which one you want.

Reply:
I think I'd prefer to just catch the error at the one place where it actually happens.

Also, can we have a test please? :)

Reply:
I've reworked the code a bit. serveListener isn't needed anymore, it can be inlined in ServeListener  which can be used directly  in the original s.serveConn. I've updated the doc too.
I don't see what test can be add value here.
What is important is notify in the changelog that this is an API breaking change for people using the quic-go http3 package, since they  cannot test/check for `quic.ErrServerClosed` anymore.